src/clr-angular/data/datagrid/datagrid-detail.spec.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Component } from '@angular/core';
+import { async } from '@angular/core/testing';
+
+import { TestContext } from './helpers.spec';
+import { ClrDatagridDetail } from './datagrid-detail';
+import { DetailService } from './providers/detail.service';
+import { createKeyboardEvent } from '../../forms/datepicker/utils/test-utils';
+import { ESC } from '../../utils/key-codes/key-codes';
+
+const content = 'Detail Pane';
+
+export default function(): void {
+  describe('ClrDatagridDetail component', function() {
+    describe('Typescript API', function() {
+      let context: TestContext<ClrDatagridDetail, FullTest>;
+      let detailService: DetailService;
+
+      beforeEach(function() {
+        context = this.create(ClrDatagridDetail, FullTest, [DetailService]);
+        detailService = context.getClarityProvider(DetailService);
+        context.detectChanges();
+      });
+
+      it('should close and hide content', () => {
```

Comment:
Does this need a test that explicetly uses an ESC event? e.g:
```
  const event = new KeyboardEvent('keyup', {
    key: 'Escape',
  });
  document.body.dispatchEvent(event);
```

Reply:
There is one below.